src/app/app.module.ts

```diff
@@ -139,6 +139,19 @@ const Services = [
                                         }
                                     },
                                 },
+                                data: {
+                                    merge(
+                                        existing: MaybeUndefined<DatasetData>,
+                                        incoming: DatasetData,
+                                        { mergeObjects },
+                                    ) {
+                                        if (existing) {
+                                            return mergeObjects(existing, incoming);
+                                        } else {
+                                            return incoming;
+                                        }
+                                    },
+                                },
```

Comment:
No difference with a merge function for `metadata` field, except types.
Use a generic function